pkg/checks/traceroute/traceroute.go

```diff
@@ -126,44 +176,69 @@ func TraceRoute(ctx context.Context, cfg tracerouteConfig) (map[int][]Hop, error
 	var wg sync.WaitGroup
 
 	for ttl := 1; ttl <= cfg.MaxHops; ttl++ {
+		c, hopSpan := tr.Start(ctx, fmt.Sprintf("Hop %d", ttl), trace.WithAttributes(
+			attribute.Int("ttl", ttl),
+		))
 		wg.Add(1)
 		go func(ttl int) {
 			defer wg.Done()
+			defer hopSpan.End()
+
 			l := log.With("ttl", ttl)
-			logctx := logger.IntoContext(ctx, l)
+			logctx := logger.IntoContext(c, l)
+
+			retry := 0
 			err := helper.Retry(func(ctx context.Context) error {
+				defer func() {
+					retry++
+				}()
+				hopSpan.AddEvent("Attempting to hop", trace.WithAttributes(
+					attribute.Int("ttl", ttl),
+					attribute.Int("retry", retry),
+				))
+
 				hop, err := traceroute(ctx, addr, ttl, cfg.Timeout)
```

Comment:
`err` redeclared here, please rename to `ierr` or something similar. So many ifs and errors here, I don't even want to spend time thinking whether we're safe here and won't just return a wrong error...   

Please enlighten me if I'm mistaken.

Reply:
Idk tbh, I've just use a new variable to be safe.

Reply:
Fine by me :+1: